pkg/http/listener.go

```diff
@@ -99,9 +101,11 @@ func ignoreErr(err error) bool {
 	return err == io.EOF
 }
 
-// start - starts separate goroutine for each TCP listener.  A valid insecure/TLS HTTP new connection is passed to httpListener.acceptCh.
-func (listener *httpListener) start() {
-	listener.acceptCh = make(chan acceptResult)
+// start - starts separate goroutine for each TCP listener.  A valid insecure/TLS HTTP new
+// connection is passed to httpListener.acceptCh, acceptch is also limited by input n number
+// of connections.
+func (listener *httpListener) start(nConn uint64) {
+	listener.acceptCh = make(chan acceptResult, nConn)
```

Comment:
httpListener works same as regular listener i.e it blocks next connection till accepted connection is passed to upper layer. This change would lead incompatible problem.  I think this limitation needs to be handled upper layer than listener